parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriteStoreV1.java

```diff
@@ -40,15 +43,23 @@
   private final int pageSizeThreshold;
   private final int dictionaryPageSizeThreshold;
   private final boolean enableDictionary;
+  private final int initialRowCountForPageSizeCheck;
+  private final boolean estimateNextSizeCheck;
   private final WriterVersion writerVersion;
   private final ByteBufferAllocator allocator;
 
   public ColumnWriteStoreV1(PageWriteStore pageWriteStore, int pageSizeThreshold, int dictionaryPageSizeThreshold, boolean enableDictionary, WriterVersion writerVersion, ByteBufferAllocator allocator) {
+    this(pageWriteStore, pageSizeThreshold, dictionaryPageSizeThreshold, enableDictionary, DEFAULT_MINIMUM_RECORD_COUNT_FOR_CHECK, DEFAULT_ESTIMATE_ROW_COUNT_FOR_PAGE_SIZE_CHECK, writerVersion, allocator);
+  }
+
+  public ColumnWriteStoreV1(PageWriteStore pageWriteStore, int pageSizeThreshold, int dictionaryPageSizeThreshold, boolean enableDictionary, int maxRowCountForPageSizeCheck, boolean estimateNextSizeCheck, WriterVersion writerVersion, ByteBufferAllocator allocator) {
     super();
     this.pageWriteStore = pageWriteStore;
     this.pageSizeThreshold = pageSizeThreshold;
     this.dictionaryPageSizeThreshold = dictionaryPageSizeThreshold;
     this.enableDictionary = enableDictionary;
+    this.initialRowCountForPageSizeCheck = maxRowCountForPageSizeCheck;
```

Comment:
Good catch.  Actually, it should be "initial" for the sig.  The issue is that v1 writer doesn't enforce a min or max and I wanted to maintain the same behavior.  So v1 will use "min" as "initial" and v2 will support "min/max".

I'll fix the name.